ShowEmptyFields.js

```diff
@@ -0,0 +1,372 @@
+    }
+    for (let i = board.x + board.x - 1; i < board.buttonId.length; i += board.x) {
+      // making array with right edges of the board(without the first one and the last one)
+      if (i < board.size - 1) {
+        this.rightEdges.push(board.buttonId[i]);
+      }
+    }
+    for (let i = 1; i < board.buttonId.length; i++) {
+      // making array with upper edges of the board(without the first one and the last one)
+      if (i < board.x - 1) {
+        this.upperEdges.push(board.buttonId[i]);
+      }
+    }
+    for (let i = board.size + 1 - board.x; i < board.buttonId.length; i++) {
+      // making array with lower edges of the board(without the first one and the last one)
+      if (i < board.size - 1) {
+        this.lowerEdges.push(board.buttonId[i]);
+      }
+    }
+    this.corners.push(
+      board.buttonId[0],
+      board.buttonId[board.x - 1],
+      board.buttonId[board.size - board.x],
+      board.buttonId[board.size - 1],
+    ); // making array with corners
+  }
+
+  getEmptyFields() { // making an array with empty fields
+    for (let i = 0; i < board.size; i++) {
+      const emptyEl = document.getElementById(board.tableId[i]);
+      if (emptyEl.innerHTML === ' ') {
+        this.emptyTabCopy.push(board.tableId[i]);
+      }
+    }
+  }
+
+  convertEmptyFieldsArrayToInt() { // converting array with empty fields into integer
+    for (let i = 0; i < this.emptyTabCopy.length; i++) {
+      let strChange = this.emptyTabCopy[i];
+      strChange = strChange.replace('td', '');
+      this.emptyTabCopy[i] = strChange;
+      this.emptyTabCopy[i] = Number(this.emptyTabCopy[i]);
+    }
+  }
+
+  getLeftEdgesOfTheBoard() {
+    let addNumToArray = 0;
+    this.leftEdgesInt.push(addNumToArray); // left edge of the board always starts with 0
+    for (let i = 0; i < board.y; i++) {
+      addNumToArray += board.x;
+      if (addNumToArray < board.size) {
+        this.leftEdgesInt.push(addNumToArray);
+      }
+    }
+  }
+
+  getRightEdgesOfTheBoard() {
+    let addNumToArray = board.x - 1;
+    this.rightEdgesInt.push(board.x - 1); // right edge of the board always starts with x-1
+    // (the count starts from 0 not from 1, so 1 should be subtracted)
+    for (let i = 0; i < board.y; i++) {
+      addNumToArray += board.x;
+      if (addNumToArray < board.size) {
+        this.rightEdgesInt.push(addNumToArray);
+      }
+    }
+  }
+
+  getFieldsWithoutLeftAndRightEdges() {
+    for (let i = 0; i < board.size; i++) {
+      this.tabsIntWithoutEdges.push(i);
+    }
+    for (let i = 0; i < this.leftEdgesInt.length; i++) {
+      const numberDelete = this.tabsIntWithoutEdges.indexOf(this.leftEdgesInt[i]);
+      this.tabsIntWithoutEdges.splice(numberDelete, 1);
+    }
+    for (let i = 0; i < this.rightEdgesInt.length; i++) {
+      const numberDelete2 = this.tabsIntWithoutEdges.indexOf(this.rightEdgesInt[i]);
+      this.tabsIntWithoutEdges.splice(numberDelete2, 1);
+    }
+  }
+
+  showFieldsWithNoMinesAround() {
```

Comment:
this function is overcomplicated. Please use recursion this will make logic much more simplier.